Objects/frameobject.c

```diff
@@ -2255,21 +2355,21 @@ PyObject*
 PyFrame_GetLocals(PyFrameObject *frame)
 {
     assert(!_PyFrame_IsIncomplete(frame->f_frame));
-    return frame_getlocals((PyObject *)frame, NULL);
+    return frame_locals_get_impl(frame);
 }
 
 PyObject*
 PyFrame_GetGlobals(PyFrameObject *frame)
 {
     assert(!_PyFrame_IsIncomplete(frame->f_frame));
-    return frame_getglobals((PyObject *)frame, NULL);
+    return frame_globals_get_impl(frame);
 }
 
 PyObject*
 PyFrame_GetBuiltins(PyFrameObject *frame)
 {
     assert(!_PyFrame_IsIncomplete(frame->f_frame));
-    return frame_getbuiltins((PyObject *)frame, NULL);
+    return frame_builtins_get_impl(frame);
```

Comment:
I think this should call `frame_builtins_get` not `frame_builtins_get_impl` otherwise this would access the frame fields without acquiring the critical section.